### DISCUSSION TITLE
I2S support

### DISCUSSION THREAD

**Discussion by MTRNord** [mciantyre/teensy4-rs#141] [Q&A] [answered]
Hi I am currently looking into doing audio with the teensy and a PCM3168APAP. This needs the usage of I2S and I2C.
I am wondering if this crate is suitable for this task or if it is a better choice to go with C and the https://github.com/PaulStoffregen/Audio lib for now :)

**Comment by mciantyre** [comment, marked as answer]
Hey there 👋 Right now, we have no I2S driver in the HAL. https://github.com/imxrt-rs/imxrt-hal/issues/19 is our I2S tracking issue. If you're interested in driver development, we'd appreciate your help there.

If that's a bit much, you could try add bindings for that audio library. Though not specifically related to audio, #122 briefly discusses ideas for integrating Rust and C++ code.

**Reply by MTRNord** [reply]
Considering I have no prior hardware dev experience I will first try the bindings path and maybe come back later to the driver :) It seems to I need to write at least an USB class anyway for my project too.
